tests/unit-core/test-mem-stats.c

```diff
@@ -26,36 +25,33 @@ const char *test_source = (
 
 int main (void)
 {
-  jerry_init (JERRY_INIT_EMPTY);
-  jerry_value_t parsed_code_val = jerry_parse ((jerry_char_t *) test_source, strlen (test_source), false);
-  TEST_ASSERT (!jerry_value_has_error_flag (parsed_code_val));
-
-  jerry_value_t res = jerry_run (parsed_code_val);
-  TEST_ASSERT (!jerry_value_has_error_flag (res));
-
-  jerry_heap_stats_t stats;
-  memset (&stats, 0, sizeof (stats));
-  bool get_stats_ret = jerry_get_memory_stats (&stats);
-  TEST_ASSERT (get_stats_ret);
-  TEST_ASSERT (stats.version == 1);
-  TEST_ASSERT (stats.size == 524280);
-
-  TEST_ASSERT (!jerry_get_memory_stats (NULL));
-
-  jerry_release_value (res);
-  jerry_release_value (parsed_code_val);
-
-  jerry_cleanup ();
-
-  return 0;
+  if (jerry_is_feature_enabled (JERRY_FEATURE_MEM_STATS))
+  {
+    jerry_init (JERRY_INIT_EMPTY);
+    jerry_value_t parsed_code_val = jerry_parse ((jerry_char_t *) test_source, strlen (test_source), false);
+    TEST_ASSERT (!jerry_value_has_error_flag (parsed_code_val));
+
+    jerry_value_t res = jerry_run (parsed_code_val);
+    TEST_ASSERT (!jerry_value_has_error_flag (res));
+
+    jerry_heap_stats_t stats;
+    memset (&stats, 0, sizeof (stats));
+    bool get_stats_ret = jerry_get_memory_stats (&stats);
+    TEST_ASSERT (get_stats_ret);
+    TEST_ASSERT (stats.version == 1);
+    TEST_ASSERT (stats.size == 524280);
+
+    TEST_ASSERT (!jerry_get_memory_stats (NULL));
+
+    jerry_release_value (res);
+    jerry_release_value (parsed_code_val);
+
+    jerry_cleanup ();
+
+    return 0;
+  }
+  else
+  {
+    return 0;
+  }
```

Comment:
The else is not necessary, call return 0 only. You can also outsource it from the if statement.